### DISCUSSION TITLE
CI benchmark fails are due to bug in todays nightly

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1358] [Announcements]
CI nightly job fails are not due to our code but due to a bug in nightly, see https://github.com/rust-lang/rust/issues/103794

Should resolve itself in a day or two.

**Comment by TheBlueMatt** [comment]
It would also be easy to swap to stable, eg we did https://github.com/lightningdevkit/rust-lightning/pull/1818

**Reply by tcharding** [reply]
We can't use stable because of `#![cfg_attr(bench, feature(test))]`, we could use another non-stable toolchain though. Cheers

**Reply by TheBlueMatt** [reply]
You can almost always use stable if `RUSTC_BOOTSTRAP=1` is in your env. Its cheating and not supported, but nor is nightly.

**Reply by tcharding** [reply]
Cool, TIL. I went digging, posting for others:

> The build system sets RUSTC_BOOTSTRAP=1. This special variable means to break the stability guarantees of rust: Allow using #![feature(...)] with a compiler that's not nightly. This should never be used except when bootstrapping the compiler.

ref: https://rustc-dev-guide.rust-lang.org/building/bootstrapping.html

**Reply by tcharding** [reply]
And this works as you say: `RUSTC_BOOTSTRAP=1 RUSTFLAGS='--cfg=bench' cargo bench`!

**Reply by Kixunil** [reply]
I'd rather pin the specific version of nightly than use this trick. They may change it in the future and our CI breaks. However I'd like to run some tests on recent nightly anyway to catch possible changes in effects of UB and to help the compiler avoid regressions. This test should be only informational and not required to  pass for merging.

**Comment by tcharding** [comment]
Issue is resolved on nightly now.
